src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java

```diff
@@ -172,6 +173,17 @@ protected boolean couldBeModifiedByMetadata(FileArtifactValue lastKnown) {
     return true;
   }
 
+  /**
+   * Optional symlink target path.
```

Comment:
Please add javadoc to explain that this is for optimizations only, maybe even rename the method to indicate that. This API should clearly state that it is an implication only: non-empty response means that this is a symlink, but the opposite may not be true.

Reply:
Done. I think the new name ("materializationExecPath") is distinctive enough to avoid confusion, and the documentation clearly states that it's only used by the prefetcher.

I'm not so sure about the second part of your comment. The artifacts that correspond to these values are *not* symlinks, as far as Bazel is concerned; they're only materialized as such in the local filesystem (which is not even a novel concept - we do the same for sandbox inputs, for example). I don't want folks to erroneously infer that they are, especially considering that there's a different kind of artifact that *is* tracked by Bazel as one.